src/os/bluestore/BlueStore.cc

```diff
@@ -14866,7 +14866,7 @@ void BlueStore::_kv_sync_thread()
       auto sync_start = mono_clock::now();
 #endif
       // submit synct synchronously (block and wait for it to commit)
-      int r = db_was_opened_read_only || cct->_conf->bluestore_debug_omit_kv_commit ?
+      int r = cct->_conf->bluestore_debug_omit_kv_commit ?
```

Comment:
This looks like a general purpose protection agains submitting txcs in read-only mode. Why removing?

Reply:
We should not run kv-sync in RO mode.
How about assert against attempts to start _kv_sync_thread when in RO mode?